### PR TITLE
Crypto-17: Integrate Navigation logic

### PR DESCRIPTION
**Pull request by vitalyaliskevych**
### Whats News
- Integrate Navigation logic
### Issues
- [Crypto-17](https://trello.com/c/ZydYq0Xg)
### Video

https://user-images.githubusercontent.com/98754902/219376512-7d9508d7-7dd1-4fa3-8274-4f4a09b811ce.mov

### CODE REVIEW

**Review by oleksandr-syurpita — APPROVED** [review]

File: Crypto/Flows/Screens/MainScreen/MainView.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import SwiftUI
 
 struct MainView: View {
-    @ObservedObject var viewModel = MainViewModel()
+    @ObservedObject var viewModel : MainViewModel
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainView.swift:11]: remove space before `:`

File: Crypto/Flows/Screens/MainScreen/MainView.swift [hunk 2]
```diff
@@ -31,17 +31,25 @@ struct MainView: View {
...
+    
+    var coinList: some View {
         VStack {
             VStack(spacing: 10) {
                 ForEach(viewModel.coinModels, id: \.id) { coinModel in
-                    createButton(coinModel: coinModel)
+                    createButton(
+                        coinModel: coinModel,
+                        action: {
+                            viewModel.navigationToDetailScreen(
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainView.swift:48]: better to have function name - `selectCoinItem` or something connected to select.
then result case should `onCoinItemSelected` for instance
and the Coordianator actually decides where we need to navigate further

File: Crypto/Flows/Screens/MainScreen/MainViewModel.swift [hunk 3]
```diff
@@ -8,6 +8,22 @@
 import Foundation
 
 class MainViewModel: ObservableObject {
+    
+    enum Result {
+        case navigationToSettingsScreen
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainViewModel.swift:13]: onSettingsSelected

File: Crypto/Flows/Screens/MainScreen/MainViewModel.swift [hunk 4]
```diff
@@ -8,6 +8,22 @@
...
+        case navigationToDetailScreen(coinName: Coin)
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainViewModel.swift:14]: `onCoinItemSelected(coin: Coin)`

File: Crypto/Flows/Screens/MainScreen/MainViewModel.swift [hunk 5]
```diff
@@ -8,6 +8,22 @@
...
+    }
+    
+    var onResult: ((Result) -> Void)?
+    
+    func navigationToSettinsScreen() {
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainViewModel.swift:19]: `selectSettings`

File: Crypto/Flows/Screens/MainScreen/MainViewModel.swift [hunk 6]
```diff
@@ -8,6 +8,22 @@
...
+        onResult?(.navigationToSettingsScreen)
+    }
+    
+    func navigationToDetailScreen(coinName: Coin) {
```
Comment (romansorochak) [inline Crypto/Flows/Screens/MainScreen/MainViewModel.swift:23]: `selectCoin(coin: Coin)`

File: Crypto/Flows/Screens/SettingsScreen/SettingsViewModel.swift [hunk 7]
```diff
@@ -9,6 +9,16 @@ import Foundation
 
 
 class SettingsViewModel: ObservableObject {
+    
     @Published var isOn = false
     
+    enum Result {
+        case navigationToMainScreen
```
Comment (romansorochak) [inline Crypto/Flows/Screens/SettingsScreen/SettingsViewModel.swift:16]: `onNavigateBack`
Here you cannot know from where this screen was actually presented

**Review by romansorochak — APPROVED** [review]

**Pull request merged**